Source/Scene/ClassificationPrimitive.js

```diff
@@ -992,8 +1157,276 @@ define([
+        points2D[3].x = -eastWestHalfDistance;
+        points2D[3].y = -northSouthHalfDistance;
+
+        // Rotate the dimensions rectangle and compute min/max in rotated space
+        var min2D = min2DScratch;
+        min2D.x = Number.POSITIVE_INFINITY;
+        min2D.y = Number.POSITIVE_INFINITY;
+        var max2D = max2DScratch;
+        max2D.x = Number.NEGATIVE_INFINITY;
+        max2D.y = Number.NEGATIVE_INFINITY;
+
+        var rotation2D = Matrix2.fromRotation(-theta, rotation2DScratch);
+        for (var i = 0; i < 4; ++i) {
+            var point2D = points2D[i];
+            Matrix2.multiplyByVector(rotation2D, point2D, point2D);
+            Cartesian2.minimumByComponent(point2D, min2D, min2D);
+            Cartesian2.maximumByComponent(point2D, max2D, max2D);
+        }
+
+        // Depending on the rotation, east/west may be more appropriate for vertical scale than horizontal
+        var scaleU, scaleV;
+        if (Math.abs(sinTheta) < Math.abs(cosTheta)) {
+            scaleU = eastWestHalfDistance / ((max2D.x - min2D.x) * 0.5);
+            scaleV = northSouthHalfDistance / ((max2D.y - min2D.y) * 0.5);
+        } else {
+            scaleU = eastWestHalfDistance / ((max2D.y - min2D.y) * 0.5);
+            scaleV = northSouthHalfDistance / ((max2D.x - min2D.x) * 0.5);
+        }
+
+        return new GeometryInstanceAttribute({
+            componentDatatype: ComponentDatatype.FLOAT,
+            componentsPerAttribute: 4,
+            normalize: false,
+            value : [sinTheta, cosTheta, scaleU, scaleV] // Precompute trigonometry for rotation and inverse of scale
+        });
+    }
+
+    var swizzleScratch = new Cartesian3();
+    function swizzle(cartesian) {
+        Cartesian3.clone(cartesian, swizzleScratch);
+        cartesian.x = swizzleScratch.z;
+        cartesian.y = swizzleScratch.x;
+        cartesian.z = swizzleScratch.y;
+        return cartesian;
+    }
+
+    var cornerScratch = new Cartesian3();
+    var northWestScratch = new Cartesian3();
+    var southEastScratch = new Cartesian3();
+    function add2DTextureCoordinateAttributes(rectangle, frameState, attributes) {
+        var projection = frameState.mapProjection;
+
+        // Compute corner positions in double precision
+        var carto = cartographicScratch;
+        carto.height = 0.0;
+
+        carto.longitude = rectangle.west;
+        carto.latitude = rectangle.south;
+
+        var corner = swizzle(projection.project(carto, cornerScratch));
+
+        carto.latitude = rectangle.north;
+        var northWest = swizzle(projection.project(carto, northWestScratch));
+
+        carto.longitude = rectangle.east;
+        carto.latitude = rectangle.south;
+        var southEast = swizzle(projection.project(carto, southEastScratch));
+
+        addAttributesForPoint(corner, 'southWest2D', attributes);
+        addAttributesForPoint(northWest, 'northWest2D', attributes);
+        addAttributesForPoint(southEast, 'southEast2D', attributes);
```

Comment:
Micro-optimization: Since these points are in the 2D plane, a lot of the values here are either duplicated or zero. So this can probably be done with 4 low + 4 high instead of the current 9 low + 9 high.

Not true for 3D case, positions are in CBF/world coordinates.